src/format/reveal/format-reveal.ts

```diff
@@ -682,6 +682,23 @@ function revealHtmlPostprocessor(
       }
     }
 
+    // https://github.com/quarto-dev/quarto-cli/issues/3533
+    // redirect anchors to the slide they refer to
+    const anchors = doc.querySelectorAll("a[href^='#/']");
+    for (const anchor of anchors) {
+      const anchorEl = anchor as Element;
+      const href = anchorEl.getAttribute("href");
+      if (href) {
+        const target = doc.getElementById(href.replace(/^#\//, ""));
+        if (target) {
+          const slide = findParentSlide(target);
+          if (slide) {
+            anchorEl.setAttribute("href", `#/${slide.getAttribute("id")}`);
```

Comment:
My only question, @jjallaire, is whether I can safely assume a slide to always have an `id`.

Reply:
In our default settings they will always have an ID but it's possible the user could create a config where they don't. I'd just do a further check on the id before setting the `href`